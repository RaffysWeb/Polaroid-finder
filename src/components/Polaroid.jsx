import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import moment from 'moment';
import shortid from 'shortid';
import { useSpring, animated } from 'react-spring';

const Polaroid = ({ item, fetchByTag }) => {
  const {
    author,
    date_taken,
    tags,
    link,
    media: { m }
  } = item;

  const props = useSpring({ from: { opacity: 0 }, to: { opacity: 1 }, config: { duration: 1000 } });

  const newTags = tags.split(' ').splice(0, 4);
  const parsedName = author.match(/(?:"[^"]*"|^[^"]*$)/)[0].replace(/"/g, '');

  return (
    <StyledPolaroid style={props}>
      <a rel="noopener noreferrer" target="_blank" href={link}>
        <ImageContainer image={m} />
      </a>
      {newTags.map(tag => (
        <button
          type="button"
          className="tag-button"
          key={shortid.generate()}
          onClick={() => fetchByTag(tag)}
        >
          #{tag}
        </button>
      ))}
      <footer>
        <p>{parsedName}</p>
        <p>{moment(date_taken).format('DD/MM/YYYY')}</p>
      </footer>
    </StyledPolaroid>
  );
};

const StyledPolaroid = styled(animated.div)`
  width: 15rem;
  height: 20rem;
  border-radius: 0.05rem;
  background-size: cover;
  background-position: center center;
  box-shadow: 0px 10px 30px -5px rgba(0, 0, 0, 0.3);
  transition: box-shadow 0.5s;
  will-change: transform;
  border: 1rem solid white;
  margin-bottom: 2rem;

  &:hover {
    position: relative;
    box-shadow: 0px 30px 100px -10px rgba(0, 0, 0, 0.7);
  }

  .tag-button {
    background: none;
    border: none;
    cursor: pointer;
    color: blue;

    &:hover {
      text-decoration: underline;
      color: darkBlue;
    }
  }

  footer {
    display: flex;
    justify-content: space-between;
    position: absolute;
    bottom: 0;
    left: 0;
    right: 0;
    p {
      margin: 0;
      font-family: 'Kaushan Script', cursive;
    }
  }
`;

const ImageContainer = styled.div`
  width: 100%;
  height: 75%;
  background-image: url(${props => props.image});
  background-size: cover;
  background-position: center;
`;

Polaroid.propTypes = {
  item: PropTypes.instanceOf(Object).isRequired,
  fetchByTag: PropTypes.func.isRequired
};

export default Polaroid;
