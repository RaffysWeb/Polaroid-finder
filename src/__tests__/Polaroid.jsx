import React from 'react';
import { create } from 'react-test-renderer';
import Polaroid from '../components/Polaroid';

describe('Polaroid tests', () => {
  const mockItem = {
    author: 'me',
    date_taken: new Date(),
    tags: 'cat',
    link: 'www.google.com',
    media: { m: 'www.google.com' }
  };

  test('Polaroid it matches the snapshot', () => {
    const component = create(<Polaroid item={mockItem} fetchByTag={jest.fn()} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
