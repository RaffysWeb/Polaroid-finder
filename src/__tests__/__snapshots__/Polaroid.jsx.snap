// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Polaroid tests Polaroid it matches the snapshot 1`] = `
<div
  className="sc-bdVaJa gNOWfD"
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <a
    href="www.google.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <div
      className="sc-bwzfXH hlYnYe"
    />
  </a>
  <button
    className="tag-button"
    onClick={[Function]}
    type="button"
  >
    #
    cat
  </button>
  <footer>
    <p>
      me
    </p>
    <p>
      31/05/2019
    </p>
  </footer>
</div>
`;
